### PR TITLE
Set minimum required version of dependencies

### PR DESCRIPTION
**Pull request by leouieda**
Establish the minimum supported version of run-time and optional dependencies (see fatiando/community#40). Support all releases from the past 2 years with a minimum of 2 minor versions. 

Needed to do some maintenance work on our CI and packaging setup to enable this:

1. Moved the requirements into separate files in the `env` folder.
2. Moved package metadata to `setup.cfg` and `pyproject.toml` so that we can programmatically convert run-time dependencies into a `requirements.txt` file with `tools/export_requirements.py`
3. Made a `tools/oldest_requirements.py` that converts `>=` into `==`. Could have been a `sed` call but getting this to work on Mac and Linux was a huge pain. The Python script is easier.
4. Simplify the CI scripts to use the requirements files and `tools` scripts to build their list of packages.

I also ended up moving `license_notice.py` to `tools` since we have more of them now.

Almost everything worked perfectly with the oldest versions out of the box. The only modification I had to make was removing some of the xxhash algorithms from the tests since they were introduced in a later version. This shouldn't be a problem because the underlying implementation is the same either way.

Also addresses fatiando/community#42

**Reminders**:

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst` and the base `__init__.py` file for the package.
- [ ] Write detailed docstrings for all functions/classes/methods. It often helps to design better code if you write the docstrings first.
- [ ] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.
- [ ] Add your full name, affiliation, and [ORCID](https://orcid.org) (optional) to the `AUTHORS.md` file (if you haven't already) in case you'd like to be listed as an author on the [Zenodo](https://zenodo.org/communities/fatiando) archive of the next release.

### CODE REVIEW

**Comment by leouieda** [comment]
@dopplershift this is largely based on our twitter chat the other day and the MetPy configuration. Would you be able to give this a quick look? No rush on this, whenever you have a spare moment.

**Review by dopplershift — COMMENTED** [review]
Looks good!

And to be clear, users don't even need to care about the `requirements.txt` thing--unless it causes installation problems due to overly tight pinning. It's something I run into while reviewing conda-forge recipes.

**Comment by leouieda** [comment]
Thanks for the feedback, Ryan! It was time we modernized our build system anyway and we've forgotten to package a `requirement.txt` file in the past so this is good to help avoid that in the future.

File: pooch/tests/test_hashes.py [hunk 1]
```diff
@@ -36,10 +36,7 @@
     "md5": "70e2afd3fd7e336ae478b1e740a5f08e",
 }
 TINY_DATA_HASHES_XXH = {
-    "xxh128": "0267d220db258fffb0c567c0ecd1b689",
-    "xxh3_128": "0267d220db258fffb0c567c0ecd1b689",
     "xxh64": "f843815fe57948fa",
-    "xxh3_64": "811e3f2a12aec53f",
     "xxh32": "98d6f1a2",
```
[the remark this answers is not in the document]
Comment (leouieda) [reply pooch/tests/test_hashes.py]: 👍🏽 good idea! Implementing that now.

**Pull request merged**